### PR TITLE
Inttegration mock server with search page

### LINKED ISSUE #98: integrate mock server with search page

### PR DESCRIPTION
**Pull request by Platton96**
Fixes #98

### CODE REVIEW

File: ecommers/lib/core/app_services/universal_app_service.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import 'package:chopper/chopper.dart';
+
+class UniversalAppService {
```
Comment (amarchenko25) [inline ecommers/lib/core/app_services/universal_app_service.dart]: remove Universal

File: ecommers/lib/core/app_services/universal_app_service.dart [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  Future<T> getDataFromBackend<T>(
```
Comment (amarchenko25) [inline ecommers/lib/core/app_services/universal_app_service.dart]: rename to fetch

File: ecommers/lib/core/app_services/universal_app_service.dart [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      Future<Response<T>> Function() quryFunction) async {
```
Comment (amarchenko25) [inline ecommers/lib/core/app_services/universal_app_service.dart]: rename to requestFunction

File: ecommers/lib/core/provider_models/search_provider_model.dart [hunk 4]
```diff
@@ -1,14 +1,50 @@
...
+
+    isBusy = false;
+  }
 
+  void clearProductsLetest() {
+    _productsLatest.clear();
     notifyListeners();
   }
+
+  Future refreshProductsRecomendation() async {
```
Comment (amarchenko25) [inline ecommers/lib/core/provider_models/search_provider_model.dart]: change recommendation to recommended

File: ecommers/lib/core/services/navigation/navigation_service.dart [hunk 5]
```diff
@@ -46,8 +46,9 @@ class NavigationService {
       case Pages.productsGrid:
         final productGridArguments = arguments as PageArguments;
         resultPage = ProductsGridPage(
-          categoryType: productGridArguments.arg1 as Categories,
-          subCategory: productGridArguments.arg2 as String,
+
```
Comment (amarchenko25) [inline ecommers/lib/core/services/navigation/navigation_service.dart]: remove blank line

File: ecommers/lib/ui/widgets/search/search_recommended_widget.dart [hunk 6]
```diff
@@ -1,34 +1,31 @@
...
+          .map((item) => createItem(item.title, context))
           .toList(),
     );
   }
 
   Widget createItem(String searchRecommendedItem, BuildContext context) {
     return InkWell(
-      //TODO: paste value to search filed
-      onTap: () => {},
+      onTap: () => navigationService.navigateTo(Pages.productsGrid),
```
Comment (amarchenko25) [inline ecommers/lib/ui/widgets/search/search_recommended_widget.dart]: navigate to product details

File: ecommers/lib/core/app_services/universal_app_service.dart (hunk 1 above)
Comment (Platton96) [reply ecommers/lib/core/app_services/universal_app_service.dart]: renamed

File: ecommers/lib/core/app_services/universal_app_service.dart (hunk 2 above)
Comment (Platton96) [reply ecommers/lib/core/app_services/universal_app_service.dart]: renamed

File: ecommers/lib/core/app_services/universal_app_service.dart (hunk 3 above)
Comment (Platton96) [reply ecommers/lib/core/app_services/universal_app_service.dart]: renamed

File: ecommers/lib/core/provider_models/search_provider_model.dart (hunk 4 above)
Comment (Platton96) [reply ecommers/lib/core/provider_models/search_provider_model.dart]: renamed

File: ecommers/lib/core/services/navigation/navigation_service.dart (hunk 5 above)
Comment (Platton96) [reply ecommers/lib/core/services/navigation/navigation_service.dart]: removed

Comment (Platton96) [reply ecommers/lib/core/services/navigation/navigation_service.dart]: removed

File: ecommers/lib/ui/widgets/search/search_recommended_widget.dart (hunk 6 above)
Comment (Platton96) [reply ecommers/lib/ui/widgets/search/search_recommended_widget.dart]: fixed

**Review by amarchenko25 — APPROVED** [review]

**Pull request merged**